### PR TITLE
[react-map-gl] Type additions and Edits.

### PR DESCRIPTION
**Pull request by pcwa-ahendricks**
Added string union type for width & height.Added string union type for width & height. Added MapLoadEvent interface for use with onLoad function.

### CODE REVIEW

**Review by pcwa-ahendricks — COMMENTED** [review]
Reviewed.

File: types/react-map-gl/index.d.ts [hunk 1]
```diff
@@ -97,8 +97,8 @@ export interface MapControllerOptions {
 }
 
 export interface ViewportProps {
-    width: number;
-    height: number;
+    width: number | string;
+    height: number | string;
```
Comment (fnberta) [inline types/react-map-gl/index.d.ts]: I don't think this is correct, these should be numbers. Take a look at https://github.com/uber/react-map-gl/blob/master/src/utils/map-state.js

**Review by fnberta — CHANGES_REQUESTED** [review]
Thanks for this contribution. The `onLoad` change looks good but I think the union of `number | string` is not correct. Happy to be convinced otherwise if I'm wrong.

File: types/react-map-gl/index.d.ts (hunk 1 above)
Comment (wheredoesyourmindgo) [reply types/react-map-gl/index.d.ts]: Thanks for the review @fnberta 

See [This example](https://github.com/uber/react-map-gl/blob/master/examples/exhibit-browserify/app.js) or (these prop-types](https://github.com/uber/react-map-gl/blob/f2bc61b6184d99c3ffa7267373b16e380dd600ba/src/components/static-map.js#L56). Using a string allows for setting width and height to '100vw' and '100vh' respectively.

Comment (fnberta) [reply types/react-map-gl/index.d.ts]: Yes, but these examples are covered by the `MapboxProps` definition which includes:
```
width: number | string;
height: number | string;
```
Or am I missing something?

Comment (pcwa-ahendricks) [reply types/react-map-gl/index.d.ts]: I see. Thanks for the clarification.

**Comment by fnberta** [comment]
I would still like to get the `onLoad` change merged. If you agree, please re-open.

**Pull request closed without merging**